### PR TITLE
Remove gauge scheduler.tasks.running

### LINKED ISSUE #29578: scheduler.tasks.running metric is always 0
### Apache Airflow version

2.5.1

### What happened

I'd expect the `scheduler.tasks.running` metric to represent the number of running tasks, but it is always zero. It appears that #10956 broke this when it removed [the line that increments `num_tasks_in_executor`](https://github.com/apache/airflow/pull/10956/files#diff-bde85feb359b12bdd358aed4106ef4fccbd8fa9915e16b9abb7502912a1c1ab3L1363). Right now that variable is set to 0, never incremented, and the emitted as a gauge.

### What you think should happen instead

`scheduler.tasks.running` should either represent the number of tasks running or be removed altogether.

### How to reproduce

_No response_

### Operating System

Ubuntu 18.04

### Versions of Apache Airflow Providers

_No response_

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by vincbeck**
As agreed in #29579, remove the gauge `scheduler.tasks.running`
closes: https://github.com/apache/airflow/issues/29578

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by eladkal — APPROVED** [review]
cc @potiuk @kaxil

**Comment by eladkal** [comment]
BTW I assume we need `{pr_number}.significant.rst` entry for this one?

File: newsfragments/30374.significant.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+The gauge ``scheduler.tasks.running`` no longer exist
+
+The gauge has never been working and its value has always been 0. Having a truthfully
+value for this metric is complex so it has been decided that removing this gauge makes
+more sense than fixing it with no certainty of the correctness of its value
```
Comment (o-nikolas) [inline newsfragments/30374.significant.rst]:
```suggestion
The gauge has never been working and its value has always been 0. Having an accurate
value for this metric is complex so it has been decided that removing this gauge makes
more sense than fixing it with no certainty of the correctness of its value.
```

**Comment by vincbeck** [comment]
Anyone to review this one? :)

**Review by o-nikolas — APPROVED** [review]
I assume that we're removing this under the pretense of a bug-fix, given that it is broken currently? If that's the case that we don't need deprecation warnings, so it lgtm.

**Review by potiuk — APPROVED** [review]

**Pull request merged**